packages/build-utils/src/transforms/README.md

```diff
@@ -0,0 +1,195 @@
+be included in the current build are found, the fences and the lines that they wrap
+are deleted. An error is thrown if a malformed fence is identified.
+
+For example, the following fenced code:
+
+```javascript
+this.store.updateStructure({
+  ...,
+  GasFeeController: this.gasFeeController,
+  TokenListController: this.tokenListController,
+  ///: BEGIN:ONLY_INCLUDE_IF(snaps)
+  SnapController: this.snapController,
+  ///: END:ONLY_INCLUDE_IF
+});
+```
+
+Is transformed as follows if the current build should not include the `snaps` feature:
+
+```javascript
+this.store.updateStructure({
+  ...,
+  GasFeeController: this.gasFeeController,
+  TokenListController: this.tokenListController,
+});
+```
+
+Note that multiple features can be specified by separating them with
+commands inside the parameter parentheses:
+
+```javascript
+///: BEGIN:ONLY_INCLUDE_IF(build-beta,build-flask)
+```
+
+### Code Fencing Syntax
+
+> In the specification, angle brackets, `< >`, indicate required tokens, while
+> straight brackets, `[ ]`, indicate optional tokens.
+>
+> Alphabetical characters identify the name and purpose of a token. All other
+> characters, including parentheses, `( )`, are literals.
+
+A fence line is a single-line JavaScript comment, optionally surrounded by
+whitespace, in the following format:
+
+```text
+///: <terminus>:<command>[(parameters)]
+
+|__| |________________________________|
+  |                  |
+  |                  |
+sentinel         directive
+```
+
+The first part of a fence line is the **sentinel** which is always the string
+"`///:`". If the first four non-whitespace characters of a line are not exactly the
+**sentinel** the line will be ignored by the parser. The **sentinel** must be
+succeeded by a single space character, or parsing will fail.
+
+The remainder of the fence line is called the **directive**
+The directive consists of a **terminus** **command** and **parameters**
+
+- The **terminus** is one of the strings `BEGIN` and `END`. It must be followed by
+  a single colon, `:`.
+- The **command** is a string of uppercase alphabetical characters, optionally
+  including underscores, `_`. The possible commands are listed later in this
+  specification.
+- The **parameters** are a string of comma-separated RegEx `\w` strings. The parameters
+  string must be parenthesized, only specified for `BEGIN` directives, and valid for its
+  command.
+
+A valid code fence consists of two fence lines surrounding one or more lines of
+non-fence lines. The first fence line must consist of a `BEGIN` directive, and
+the second an `END` directive. The command of both directives must be the same,
+and the parameters (if any) must be valid for the command. Nesting is not intended
+to be supported, and may produce undefined behavior.
+
+If an invalid fence is detected, parsing will fail, and the transform will throw
+an error.
+
+### Commands
+
+#### `ONLY_INCLUDE_IF`
```

Comment:
Added to PR description:

> - `ONLY_INCLUDE_IN` has been renamed to `ONLY_INCLUDE_IF`, which is more in line with its actual usage.
>  - When code fencing was originally implemented, we imagined that it would be used to differentiate between specific builds, e.g. "main" and "beta". Since then, the parameters of the command are labels corresponding to different features. Thus, fenced code is not included _in_ certain builds so much as _if_ certain features are enabled.